native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/internal/NativeImageCommandLineProvider.java

```diff
@@ -108,24 +106,33 @@ private List<String> buildExcludeConfigArgs(NativeImageOptions options) {
         List<String> args = new ArrayList<>();
         options.getExcludeConfig().get().forEach((dependency, listOfResourcePatterns) -> {
             // Resolve jar for this dependency.
-            project.getConfigurations().getByName("runtimeClasspath").getIncoming().artifactView(view -> {
-                view.setLenient(true);
-                view.componentFilter(id -> {
-                    if (id instanceof ModuleComponentIdentifier) {
-                        ModuleComponentIdentifier mid = (ModuleComponentIdentifier) id;
-                        String gav = String.format("%s:%s",
-                                mid.getGroup(),
-                                mid.getModule()
-                        );
-                        return dependency.startsWith(gav);
-                    }
-                    return false;
-                });
-            }).getFiles().forEach(jarPath -> listOfResourcePatterns.forEach(resourcePattern -> {
-                args.add("--exclude-config");
-                args.add(jarPath.toPath().toAbsolutePath().toString());
-                args.add(String.format("\"%s\"", resourcePattern));
-            }));
+            options.getClasspath()
+                    .getFrom().stream()
+                    .filter(Configuration.class::isInstance)
+                    .map(Configuration.class::cast)
+                    .map(configuration -> configuration.getIncoming().artifactView(view -> {
+                        view.setLenient(true);
+                        view.componentFilter(id -> {
+                            if (id instanceof ModuleComponentIdentifier) {
+                                ModuleComponentIdentifier mid = (ModuleComponentIdentifier) id;
+                                String gav = String.format("%s:%s:%s",
+                                        mid.getGroup(),
+                                        mid.getModule(),
+                                        mid.getVersion()
+                                );
+                                return gav.startsWith(dependency);
+                            }
+                            return false;
+                        });
+                    }).getFiles().getFiles())
+                    .flatMap(Set::stream) // merge resulting sets of files
```

Comment:
:x: You can use `.distinct()` and I would collect to a `List` so that the order is reproducible.